test/unit/services/IssueService.test.js

```diff
@@ -0,0 +1,211 @@
+const { expect } = require("chai");
+const { stub } = require("sinon");
+
+const Issue = require("../../../models/Issue");
+const issueService = require("../../../services/IssueService");
+
+describe("IssueService", () => {
+  // Todo: Add tests for getAllIssues() & getAllIssuesByDepartment()
+  //? Not getting how to test the chained mongoose methods
+
+  describe("getIssueById(id)", async () => {
+    let findOne;
+    beforeEach(() => {
+      findOne = stub(Issue, "findOne");
+    });
+    afterEach(() => findOne.restore());
+
+    it("Resolves if Issue.findOne resolves", async () => {
+      findOne.resolves({});
+      try {
+        const issue = await issueService.getIssueById(1);
+        expect(findOne.calledOnce).to.be.true;
+        expect(issue).to.be.empty;
+      } catch (error) {
+        expect(error).to.be.undefined;
+      }
+    });
+
+    it("Rejects if Issue.findOne rejects", async () => {
+      findOne.rejects(new Error("No issue."));
+      try {
+        const issue = await issueService.getIssueById(1170254);
+        expect(issue).to.be.null;
+        expect.fail("Must return NULL");
+      } catch (error) {
+        expect(error.message).to.be.equal("No issue.");
+      }
+    });
+  });
+
+  describe("addIssue()", () => {
+    const newIssue = {
+      name: "Test",
+      description: "description",
+      section: "section",
+      department: "department",
+      isEdited: false,
+      isResolved: false,
+      isInappropriate: false,
+      upvotes: 0,
+      upvoters: [],
+      comments: [],
+      solutions: [],
+      createdBy: "userID",
+      createdOn: new Date().toLocaleString(),
+    };
+
+    let create;
+    beforeEach(() => {
+      create = stub(Issue, "create");
+    });
+    afterEach(() => create.restore());
+
+    it("Resolves if Issue.create resolves", async () => {
+      create.resolves(newIssue);
+      try {
+        const issue = await issueService.addIssue(
+          "Test",
+          "description",
+          "section",
+          "department",
+          "userID"
+        );
+        expect(create.calledOnce).to.be.true;
+        expect(issue).to.equal(newIssue);
```

Comment:
This should fail as `createdBy` will not have anything in `issue` as the `typecasting` will fail.

Reply:
Will definitely look into it